onnxscript/tests/function_libs/torch_lib/ops_test_common.py

```diff
@@ -355,6 +355,41 @@ def _format_model_and_input_information(onnx_model, inputs):
     )
 
 
+TORCH_DTYPE_TO_ONNX_STRING = {
+    torch.bool: "tensor(bool)",
+    torch.uint8: "tensor(uint8)",
+    torch.int8: "tensor(int8)",
+    torch.int16: "tensor(int16)",
+    torch.int32: "tensor(int32)",
+    torch.int64: "tensor(int64)",
+    torch.float16: "tensor(float16)",
+    torch.float32: "tensor(float)",
+    torch.float64: "tensor(double)",
+    torch.complex64: "tensor(complex64)",
+    torch.complex128: "tensor(complex128)",
+    torch.bfloat16: "tensor(bfloat16)",
+}
+
+
+def dtype_op_schema_compatible(dtype: torch.dtype, schema: onnx.defs.OpSchema) -> bool:
+    if not schema.inputs:
+        # If there are no inputs, we can't check compatibility. Assume it is compatible.
+        # e.g. aten_randn has only attributes.
+        return True
+    if schema.inputs[0].name not in {"self", "input"}:
+        # If the first input is not self or input, it is usually another input that is not
+        # the same type as the output. Assume it is compatible.
+        # e.g. aten_add has inputs "self" and "other".
+        return True
+    first_input_dtype = schema.inputs[0].type_str
```

Comment:
Related to the comment on this function, I'm not quite following the logic of below lines. Could you briefly explain?

Reply:
Thoughts are clearer after realizing `first_input_dtype` is a type param str like `TReal`. Is it guaranteed that `schema.inputs[0].type_str` is always a type param str?

Reply:
> Is it guaranteed that schema.inputs[0].type_str is always a type param str?

Do you mean it can be `sequence(tensor(int64))`? I realized that's a case I didn't handle.

Reply:
But it's also just ops like `def aten_stack(tensors: Sequence[TTensorOrString], dim: int = 0)` whose first input is name "self" or "input" so we always test it on all dtypes.

Reply:
Cool for finding an edge case, but to your question slightly no, I was just wondering if it is legal to set `tensor(float32)` for `schema.inputs[0].type_str`.  I think it shouldn't, otherwise it becomes weird for `schema.type_constraints`.

Reply:
Or maybe it's legal when that's the only type supported, and type_constraints could be empty?

Reply:
type_str will be a name (`T1`, `TFloat` etc.) We use this name to find the allowed types in type_constraints

Reply:
That's guaranteed to not be `tensor(float32)` if that was the question

Reply:
> if it is legal to set `tensor(float32)` for `schema.inputs[0].type_str`. 

It is not. Or I should say we _can_ (we may be able to use it as a name), but it still needs to correspond to an entry in type_constraints. Otherwise there will an initialization error in OpSchema I believe.